crnk-core/src/main/java/io/crnk/core/engine/internal/document/mapper/DocumentMapper.java

```diff
@@ -143,7 +145,8 @@ private LinksInformation enrichSelfLink(LinksInformation linksInformation, Query
 							linksInformation = selfLinksInformation;
 						}
 
-						JsonApiUrlBuilder.UrlParameterBuilder urlBuilder = new JsonApiUrlBuilder.UrlParameterBuilder(requestUri.toString());
+						JsonApiUrlBuilder.UrlParameterBuilder urlBuilder = new JsonApiUrlBuilder.UrlParameterBuilder(UrlUtils.concat(queryAdapter.getQueryContext().getBaseUrl(),
```

Comment:
what isnwring with the request uri? maybe a fix should be applied there if necessary. currently the code is inconsitent, having null check for resourceuri, but does not make use it.